### PR TITLE
Bundle JS files using Webpack

### PR DESCRIPTION
**Pull request by dtognazzini**
These changes usher in using Webpack for bundling. The end goal is to allow consumers to import Backdraft source files directly. For now, Webpack is used to build the distribution and for transpiling a la babel-loader.

This PR partially addresses #84.

Source files were modified only to the extent necessary to get Webpack working.

The following issues were filed for follow up work:
#87 - Use Browserlist?
#88 - Declare Dependencies 
#89 - Hook up Hound for ESLint. 
#90 - Revamp source to allow consumers to pull in files directly.
#91 - Overhaul specs to use ES6 modules.

### CODE REVIEW

File: src/plugins/data_table/dataTables.colReorder.js [hunk 1]
```diff
@@ -25,6 +25,10 @@
  * Minor bug fixes by Jeremy Hubble @jeremyhubble
  */
 
+import jQuery from "jquery";
+import {Coordinates, DomVisibility} from "../../utils/dom_visibility";
```
Comment (garvank) [inline src/plugins/data_table/dataTables.colReorder.js]: Spaces inside `{}`'s 🙃

File: src/plugins/data_table/data_table.js [hunk 2]
```diff
@@ -1,6 +1,21 @@
...
+
+import ColumnManager from "./column_manager";
+import Plugin from "../../plugin";
+
+import SelectionManager from "./selection_manager";
+import LockManager from "./lock_manager";
+
+import DataTableFilter from "./filter_view";
+
+import {extractColumnCSSClass, toColumnCSSClass} from "../../utils/css";
```
Comment (garvank) [inline src/plugins/data_table/data_table.js]: Same comment re:`{}`

**Comment by dtognazzini** [comment]
@garvank Thanks for the review!

> One question: Do you think it would be valuable to set src/utils as a modules directory? Could be nice to just import the Util you need without referencing it directly.

I'm not sure what you mean. Let's talk in person.

Regarding the style comments, my inclination is to forego making those changes for now and instead leave them to be addressed by a new task that hooks up JSLint for this repo. We will want to use JSLint in `web/` at some point. We can leverage learnings from using it here when we use it in web.

File: package.json [hunk 3]
```diff
@@ -16,18 +15,26 @@
...
-    "babel-preset-es2016": "^6.24.1",
-    "babel-preset-minify": "^0.2.0",
-    "babel-plugin-transform-es2015-modules-systemjs": "^6.24.1"
+    "underscore": "1.8.3",
+    "webpack": "^3.10.0"
+  },
+  "peerDependencies": {
+    "backbone": "1.1.2",
+    "jquery": "^1.10.2",
+    "underscore": "1.8.3"
```
Comment (nburwell) [inline package.json]: nice!

File: src/backdraft.js [hunk 4]
```diff
@@ -1,30 +0,0 @@
...
-  {%= inline("src/plugin.js") %}
-  Backdraft.plugin = Plugin.factory;
-
-  {%= inline("src/plugins/base/base.js") %}
-  {%= inline("src/plugins/data_table/index.js") %}
-  {%= inline("src/plugins/listing/index.js") %}
-
-  window.Backdraft = Backdraft;
-
-})(jQuery);
```
Comment (nburwell) [inline src/backdraft.js]: so glad to see this gone!

**Comment by dtognazzini** [comment]
@nburwell 
> are the environment setup instructions still correct or should it change to yarn install etc?

No they are not. I'll add another commit for that. Thanks!

**Comment by nburwell** [comment]
> No they are not. I'll add another commit for that.
Sweet thanks

👍

**Comment by nburwell** [comment]
👍 on last changes

**Pull request merged**